rasa/nlu/extractors/entity_synonyms.py

```diff
@@ -44,59 +68,62 @@ def train(
                 entity_val = example.get(TEXT)[entity["start"] : entity["end"]]
                 self.add_entities_if_synonyms(entity_val, str(entity.get("value")))
 
-    def process(self, message: Message, **kwargs: Any) -> None:
+        if self._resource:
+            self._persist()
 
-        updated_entities = message.get(ENTITIES, [])[:]
-        self.replace_synonyms(updated_entities)
-        message.set(ENTITIES, updated_entities, add_to_output=True)
+    def process(self, messages: List[Message]) -> None:
```

Comment:
we need to return the processed message even though they are modified in place to make it explicit that the graph node passes an output into the next component (otherwise the data flow would be implicit again)
```suggestion
    def process(self, messages: List[Message]) -> List[Message]:
```